clang/lib/CodeGen/CGBuilder.h

```diff
@@ -154,6 +154,18 @@ class CGBuilderTy : public CGBuilderBaseTy {
                                             Ordering, SSID);
   }
 
+  llvm::Value *CreateBitCast(llvm::Value *Ptr, llvm::Type *Ty,
+                             const llvm::Twine &Name = "") {
+    if (Ty->isPointerTy() &&
+        Ptr->getType()->getPointerAddressSpace() !=
+            Ty->getPointerAddressSpace()) {
+      Ty = llvm::PointerType::getWithSamePointeeType(
+          llvm::cast<llvm::PointerType>(Ty),
+          Ptr->getType()->getPointerAddressSpace());
+    }
+    return CGBuilderBaseTy::CreateBitCast(Ptr, Ty, Name);
```

Comment:
That's not legal

see https://llvm.org/docs/LangRef.html#bitcast-to-instruction

```
Pointer (or vector of pointers) types may only be converted to other pointer (or vector of pointers) types with the same address space through this instruction.
```

Reply:
@jyu2-git  Could you please check the implementation?  It's copied from your patch.

Reply:
if I understand correctly given your other patch, you only care about the x86 backend. If so, I suspect your issue relates to the address space map or a component that (wrongly) emits an hard coded address space. But as there is no test, this hard to understand why you want to do this.

Reply:
We use the clang to compile the following piece of OpenCL kernel code.  The code has dependency on one close source header opencl-intrin-wrapper.h I cannot provided here. And the compilation reports cast invalid error if we do not add the code in this PR. 

#define __OPENCL__
#include "opencl-intrin-wrapper.h"

int16 __ocl_imasked_load_int16(__private int16 *pLoadAdd, ushort mask)
{
  return as_int16(_mm512_maskz_loadu_epi32((__mmask16) mask, (void const *)pLoadAdd));
}

Reply:
Right, so I suppose you have some kind of mixture in your compiler invocation that uses SPIR and X86 (normally pretty much everything default to 0 for x86 but you would get issue with mangling).

IMO, that looks like you need an address space cast to stay in line with llvm semantic.